Runtime/Scripts/DataCollection/RGDataCollection.cs

```diff
@@ -0,0 +1,135 @@
+
+            // Setup all listener-based data collection (as opposed to tick-based)
+            Application.logMessageReceivedThreaded += CaptureLog;
+
+            Debug.Log("RGSession created and started!");
+        }
+
+        public void CaptureScreenshot(long tick)
+        {
+            Debug.Log($"Captured screenshot at tick {tick}");
+            string path = GetSessionDirectory($"screenshots/{tick}.jpg");
+
+            var texture = ScreenCapture.CaptureScreenshotAsTexture(1);
+
+            // Encode the texture into a jpg byte array
+            byte[] bytes = texture.EncodeToJPG(100);
+
+            // Save the byte array as a jpg file
+            File.WriteAllBytes(path, bytes);
+
+            // Destroy the texture to free up memory
+            Object.Destroy(texture);
+        }
+
+        public void CaptureReplayDataPoint(RGTickInfoData tickInfo, long playerId, RGActionRequest[] actions, RGValidationResult[] validations)
+        {
+            _replayDataForTick[tickInfo.tick] = new ReplayDataPoint(tickInfo, playerId, actions, validations);
+        }
+
+        public void CaptureLog(string logString, string stackTrace, LogType type)
+        {
+            _logDataPoints.Enqueue(new LogDataPoint(logString, stackTrace, type));
+        }
+
+        public void RecordSession(long botInstanceId, RGClientConnectionType rgClientConnectionType)
+        {
+            Debug.Log("Ending RGSession...");
+            
+            Debug.Log("Saving all replay data files and zipping...");
+            foreach (var replayDataPoint in _replayDataForTick)
+            {
+                var replayData = JsonConvert.SerializeObject(replayDataPoint.Value, Formatting.None, new JsonSerializerSettings
+                {
+                    ReferenceLoopHandling = ReferenceLoopHandling.Ignore
+                });
+                File.WriteAllText(GetSessionDirectory($"replay_data/{replayDataPoint.Key}.txt"), replayData);
+            }
+            // Zip the replay data text files
+            
+
+            // Now, begin uploading
+            Debug.Log($"Uploading data collected for bot {botInstanceId}...");
+            
+            // First, upload all of the screenshots
+            var screenshotFiles = Directory.GetFiles(GetSessionDirectory($"screenshots"));
+            foreach (var screenshotFilePath in screenshotFiles)
+            {
+                var screenshotFile = Path.GetFileName(screenshotFilePath);
+                var tick = long.Parse(screenshotFile.Split(".")[0]);
+                RGServiceManager.GetInstance()?.UploadScreenshot(
+                    botInstanceId, tick, screenshotFilePath,
+                    () =>
+                    {
+                        Debug.Log($"Successfully uploaded screenshot for bot instance {botInstanceId} and tick {tick}");
+                    },
+                    () =>
+                    {
+                        
+                    });
```

Comment:
This might be good to wrap in a `try/catch`. If an upload fails, we can still try the others

Reply:
Nevermind. It looks like the `UploadScreenshot` function handles that. I'll leave this comment here for fun